### ISSUE TITLE
Flaky test in test_precision.py

### ISSUE THREAD

**Issue by loopylangur** [OpenMined/PySyft#2937]
**Describe the bug**
The test `test_torch_sigmoid_approx` in `test/torch/tensors/test_precision.py` fails intermittently (4 out of 30 times that i tried) with the following error:

```python
>               assert (diff / (tolerance * norm)) < 1  
E               assert (tensor(0.0585) / (0.05 * tensor(0.9880))) < 1  
```
Does this indicate a bug or a dependency issue? I can help contribute to a fix if suggested.

**To Reproduce**
The error can be reproduced using the following seed:
```
def test_torch_sigmoid_approx(workers):
       torch.manual_seed(7893584826795948387)
       .....
```

**Expected behavior**
The test should always pass

**Screenshots**
NA

**Desktop (please complete the following information):**
Environment:
```
Ubuntu 16.04
python 3.7
numpy==1.17.4
syft-tensorflow==0.1.0
tensorflow==1.15.0
torch==1.3.0
torchvision==0.4.1 
```

**Additional context**
Full error log:
```
Falling back to insecure randomness since the required custom op could not be found for the installed version of TensorFlow. Fix this by compiling custom ops. Missing file was 'python3.7/site-packages/tf_encrypted/operations/secure_random/secure_random_module_tf_1.15.0.so'
WARNING:tensorflow:From python3.7/site-packages/tf_encrypted/session.py:24: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.6, pytest-5.3.2, py-1.8.1, pluggy-0.13.1 -- bin/python
cachedir: .pytest_cache
rootdir: PySyft, inifile: setup.cfg
collecting ... collected 1 item

test/torch/tensors/test_precision.py::test_torch_sigmoid_approx FAILED

==================================================================================================== FAILURES =====================================================================================================
____________________________________________________________________________________________ test_torch_sigmoid_approx ____________________________________________________________________________________________

workers = {'alice': <VirtualWorker id:alice #objects:580>, 'bob': <VirtualWorker id:bob #objects:544>, 'charlie': <VirtualWorker id:charlie #objects:0>, 'james': <VirtualWorker id:james #objects:81>, ...}

    @assert_time(max_time=40)
    def test_torch_sigmoid_approx(workers):
        """
        Test the approximate sigmoid with different tolerance depending on
        the precision_fractional considered
        """
        #np.random.seed(771475754)
        torch.manual_seed(7893584826795948387)
        alice, bob, james = workers["alice"], workers["bob"], workers["james"]
    
        fix_prec_tolerance_by_method = {
            "exp": {3: 5 / 100, 4: 1 / 100, 5: 1 / 100},
            "maclaurin": {3: 7 / 100, 4: 15 / 100, 5: 15 / 100},
        }
    
        for method, fix_prec_tolerance in fix_prec_tolerance_by_method.items():
            for prec_frac, tolerance in fix_prec_tolerance.items():
                t = torch.tensor(range(-10, 10)) * 0.5
                t_sh = t.fix_precision(precision_fractional=prec_frac).share(
                    alice, bob, crypto_provider=james
                )
                r_sh = t_sh.sigmoid(method=method)
                r = r_sh.get().float_prec()
                t = t.sigmoid()
                diff = (r - t).abs().max()
                norm = (r + t).abs().max() / 2
    
>               assert (diff / (tolerance * norm)) < 1
E               assert (tensor(0.0585) / (0.05 * tensor(0.9880))) < 1

test/torch/tensors/test_precision.py:474: AssertionError
================================================================================================ 1 failed in 5.91s ================================================================================================

```

**Comment by gmuraru** [comment]
@loopylangur could you test this after #2998 gets merged?

**Comment by gmuraru** [comment]
@loopylangur the PR got merged - could you retest this, please?

**Comment by loopylangur** [comment]
Hi, I can still reproduce the error with the latest version.
